SharedSequence/src/test/kotlin/org/notests/sharedsequence/utils/TestSchedulerRule.kt

```diff
@@ -0,0 +1,38 @@
+package org.notests.sharedsequence.utils
+
+import io.reactivex.Scheduler
+import io.reactivex.android.plugins.RxAndroidPlugins
+import io.reactivex.internal.schedulers.ExecutorScheduler
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.TestScheduler
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+/**
+ * Created by juraj begovac on 13/9/18
+ */
+
+class TestSchedulerRule : TestRule {
```

Comment:
What is this Rule thing doing?

Reply:
Replaces `mainScheduler` (and all other schedulers) with `testScheduler`. It's because I cannot test Driver without it because Driver uses `AndroidSchedulers.mainThread()`. And I cannot use mockito for making something like this 
`when(Driver.scheduler).theReturn(testScheduler)` because this is static method in Driver. Maybe I can use powermock to mock static methods but I found this rule more suitable. And I'm open for better soultions :)

Reply:
Oh I see. Can you add a coment or a link to some documentation on this? It's not easy to understand what's going on. Especially the `immediate` scheduler. How are you able to pass in `Runnable::run` as an parameter of `ExecturoScheduler`?